.github/workflows/ci.yml

```
@@ -30,6 +30,37 @@ jobs:
           context: .
           profile-name: "mpc-image-builder"
           file: deployment/Dockerfile-mpc
+          tags: test_image_tag_ci
+          load: true
+
+      - name: Check Docker image can initialize and start
+        run: |
+          touch /tmp/image-digest.bin
+          # Test container startup - fail if container can't start
+          timeout -k 10 120s docker run --rm \
+          -e MPC_HOME_DIR="/tmp" \
+          -e MPC_ACCOUNT_ID=test_image.near \
+          -e RUST_BACKTRACE="full" \
+          -e RUST_LOG="mpc=debug,info" \
+          -e MPC_SECRET_STORE_KEY=BD399143F5B3126098B0EAA023A0E730 \
+          -e MPC_P2P_PRIVATE_KEY=ed25519:2WBi5gRyJntYA7dCyddiwNV2yNTdr5uZhrP4WX5GNBL5DSVCSR7ESvcXF2DBfY5oPYhzBmHnguPVSXjr6UCi8h1g \
+          -e MPC_ACCOUNT_SK=ed25519:6X3Bnghdf89WjHQfHDFYjW2UeNaNqCQ1AUGxX7zgvbJT4KSoeXzuHEdux6A2jsphTZTmS4SUGQRyYqC2ik3UrMP \
+          -e NEAR_BOOT_NODES=ed25519:ERguu7jQuYk8pxNsRC6FdezvNsegBPva1GRGqjmtD7i2@10.10.10.10:24567 \
+          -e MPC_CONTRACT_ID=v1.signer \
+          -e MPC_IMAGE_HASH=5ba283860c0efa3d4c3e08a76a2b77fab4725baad4f48504eac858e04af7fd64 \
+          -e MPC_LATEST_ALLOWED_HASH_FILE=/tmp/image-digest.bin \
+          -e MPC_ENV=mainnet test_image_tag_ci
```

It would be more resilient to use localnet here imo. This way we don't accidentally try to connect to mainnet nodes.

```suggestion
          -e MPC_ENV=localnet test_image_tag_ci
```